subsys/bluetooth/host/att.c

```diff
@@ -357,65 +409,64 @@ static bt_att_chan_sent_t chan_cb(struct net_buf *buf)
 	case ATT_REQUEST:
 	case ATT_INDICATION:
 		return chan_req_sent;
+	case ATT_COMMAND:
+	case ATT_NOTIFICATION:
+		return chan_tx_complete;
 	default:
-		return NULL;
+		__ASSERT(false, "Unknown op type");
 	}
+	CODE_UNREACHABLE;
```

Comment:
Without it GCC complains about not returning anything from the function.

Reply:
Ah, because it is missing a return value in the `default` case of course. In the case where `CONFIG_ASSERT=n`, this line is no longer unreachable, is it?

Reply:
It would be reachable if asserts are compiled out, yes.

Reply:
Does it need any change, or is this OK to use like this? @alwa-nordic @jhedberg ?

Reply:
Add a return statement into the default branch after the assert instead

Reply:
Added a callback and return so that `tx_meta_free()` is called even if the host sends an invalid op code.